vm/src/stdlib/itertools.rs

```diff
@@ -73,6 +73,66 @@ impl PyItertoolsChain {
     }
 }
 
+#[pyclass(name = "compress")]
+#[derive(Debug)]
+struct PyItertoolsCompress {
+    data: PyObjectRef,
+    selector: PyObjectRef,
+}
+
+impl PyValue for PyItertoolsCompress {
+    fn class(vm: &VirtualMachine) -> PyClassRef {
+        vm.class("itertools", "compress")
+    }
+}
+
+#[pyimpl]
+impl PyItertoolsCompress {
+    #[pymethod(name = "__new__")]
+    #[allow(clippy::new_ret_no_self)]
+    fn new(
+        _cls: PyClassRef,
+        data: PyObjectRef,
+        selector: PyObjectRef,
+        vm: &VirtualMachine,
+    ) -> PyResult {
+        let data_iter = get_iter(vm, &data)?;
+        let selector_iter = get_iter(vm, &selector)?;
+
+        Ok(PyItertoolsCompress {
+            data: data_iter,
+            selector: selector_iter,
+        }
+        .into_ref(vm)
+        .into_object())
+    }
+
+    #[pymethod(name = "__next__")]
+    fn next(&self, vm: &VirtualMachine) -> PyResult {
+        let selector_iter = get_iter(vm, &self.selector)?;
+        let sel_obj = call_next(vm, &selector_iter)?;
+
+        let verdict = objbool::boolval(vm, sel_obj.clone())?;
+
+        let data_iter = get_iter(vm, &self.data)?;
+        let data_obj = call_next(vm, &data_iter);
```

Comment:
You may want to add the question-mark operator to the end of this line, like this:
```rust
let data_obj = call_next(vm, &data_iter)?;
```
What happens, is that in case of `Err` we return from the function, in case of `Ok`, we get the value out.